src/components/Question.tsx

```diff
@@ -0,0 +1,101 @@
+import { useMutation } from "@tanstack/react-query";
+import { useState } from "react";
+import { createVote } from "../client/VotingClient";
+import { VoteQuestion, VoteSelection } from "../util/Types";
+
+interface QuestionProps {
+  question: VoteQuestion;
+  member_id: string | undefined;
+}
+
+export const Question = ({ question, member_id }: QuestionProps) => {
+  const [selectedVote, setSelectedVote] = useState<VoteSelection>();
+  const [error, setError] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
+
+  const availableOptions = [
+    { value: VoteSelection.YES, label: "Yes" },
+    { value: VoteSelection.NO, label: "No" },
+    { value: VoteSelection.ABSTAIN, label: "Abstain" },
+  ];
+
+  const getLabelFromSelection = (value: VoteSelection | undefined) => {
+    return availableOptions.find((option) => option.value === value)?.label;
+  };
+
+  const createVoteMutation = useMutation({
+    mutationFn: async (selectedVote: VoteSelection) => {
+      return createVote(question.uuid, member_id, selectedVote);
+    },
+    onSuccess: () => {
+      setSubmitted(true);
+      // TODO: this success should update the voting history query however
```

Comment:
what do you mean by this? could we just invalidate the voting history query?

Reply:
Whenever a vote is successfully created it should update the content on the vote record page ( your vote and submssion). The useMutation can force an endpoint to refresh using invalidate query, so I was trying get the endpoint that fetches the votingHistory to refresh, but adding that in was essentially skipping the confirmation message shown at the end of the vote, which isn't ideal.